cumulusci/core/flowrunner.py

```diff
@@ -628,33 +630,37 @@ def _check_old_yaml_format(self):
             else:
                 raise FlowConfigError("No steps found in the flow definition")
 
-    def _check_infinite_flows(self, flow_config, visited_flows=None):
+    def _check_infinite_flows(self, flow_config, flow_stack=None):
         """
         Recursively loop through the flow_config and check if there are any cycles.
 
-        :param steps: Set of step definitions to loop through
-        :param flows: Flows already visited.
+        :param flow_config: FlowConfig to traverse to find cycles/infinite loops
+        :param flow_stack: list of flow signatures already visited
         :return: None
         """
-        if visited_flows is None:
-            visited_flows = set()
+        if not flow_stack:
+            flow_stack = []
         project_config = flow_config.project_config
+
         for step in flow_config.steps.values():
             if "flow" in step:
                 flow_name = step["flow"]
```

Comment:
It might be helpful to name this to `next_flow_name` or `sub_flow_name` to indicate this is the different than the flow we are looping through.

Reply:
That's a good idea 👍🏽